src/features/cseMachine/components/ArrayNullUnit.tsx

```diff
@@ -1,18 +1,18 @@
 import React from 'react';
 import { Line as KonvaLine } from 'react-konva';
 
-import CseMachine from '../CseMachine';
 import { Config, ShapeDefaultProps } from '../CseMachineConfig';
 import { Layout } from '../CseMachineLayout';
 import { ReferenceType } from '../CseMachineTypes';
+import { defaultSAColor } from '../CseMachineUtils';
 import { Visible } from './Visible';
 
 /** this classes encapsulates a null value in Source pairs or arrays */
 export class ArrayNullUnit extends Visible {
-  arrayUnit: ReferenceType;
-  referencedBy: ReferenceType[];
+  arrayUnit: ReferenceType | ReferenceType;
+  referencedBy: (ReferenceType | ReferenceType)[];
 
-  constructor(referencedBy: ReferenceType[]) {
+  constructor(referencedBy: (ReferenceType | ReferenceType)[]) {
```

Comment:
`ReferenceType | ReferenceType` is just `ReferenceType`. Could you clean this up? Thanks!